### PR TITLE
fix: eth_getTransactionByHash + eth_getTransactionReceipt fixes

### PR DESCRIPTION
**Pull request by raulk**
## Related Issues

- Hash was 0, thus leading to issues with Hardhat and other tools.
- BlockHash, BlockNumber, and TransactionIndex must be null (not zero) when tx is returned from mpool.
- Transaction receipt with no logs should return an empty logs array (instead of null).
- For correctness, a transaction receipt with no logs should return a 256 bytes sized bloom.
- Return nonce.

## Proposed Changes

Fix these issues and add coverage in FEVM integration tests.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [X] Commits have a clear commit message.
- [X] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: itests/eth_deploy_test.go
Comment (arajasek) [inline itests/eth_deploy_test.go:105]: Yeah, let's not play this game, we're inviting flakiness here.

Can we just try to lookup the receipt before making these assertions? If found, let's simply skip them (with a log saying that we're doing so).

File: itests/eth_deploy_test.go
Comment (arajasek) [inline itests/eth_deploy_test.go:33]: I'd consider increasing this -- this test isn't really constrained by epochs (needs ~3 epochs), so even if the blocktime were like 5 seconds, it would still finish pretty quickly

File: itests/eth_deploy_test.go
Comment (arajasek) [inline itests/eth_deploy_test.go:116]: Let's just actually `StateWaitMsg` here instead of this funkiness -- if we don't have the tooling to easily get the msgCid, slap a TODO on it for now (I think the work @geoff-vball  is doing will make this easy to do).

**Review by arajasek — APPROVED** [review]
Approved if we want a quick merge, but would like to see the suggested changes to the test implemented (can wait until we do the feat/nv18-fevm -> master merge if necessary).

File: itests/eth_deploy_test.go
Comment (raulk) [reply itests/eth_deploy_test.go:105]: This is actually a pretty important assertion (we had bugs here before, so want to lock these for regression testing). I'm onboard with deflaking, as long as we don't withdraw coverage we need.

File: itests/eth_deploy_test.go
Comment (raulk) [reply itests/eth_deploy_test.go:33]: Plus the boot up time for the chain (around 10 epochs).

File: itests/eth_deploy_test.go
Comment (raulk) [reply itests/eth_deploy_test.go:116]: We do need to check that the receipt is null while the transaction hasn't landed on chain -- it's part of the test's coverage.

**Comment by raulk** [comment]
In case you're wondering if I'm making up the how this API should behave, I am not. Sources include:

- https://docs.infura.io/infura/networks/ethereum/json-rpc-methods/eth_gettransactionbyhash
- https://ethereum.org/es/developers/docs/apis/json-rpc/
- Alchemy docs

File: itests/eth_deploy_test.go
Comment (geoff-vball) [reply itests/eth_deploy_test.go:105]: The Miners in the itest framework have `pause` and `unpause` functions exposed, but we should implement a function for the test kit that allows us to pause and unpause all mining.

File: itests/eth_deploy_test.go
Comment (arajasek) [reply itests/eth_deploy_test.go:116]: Sure, keep doing that every 500ms, but the cue to stop is StateWaitMsg returning (not "i waited 10 seconds").

Comment (raulk) [reply itests/eth_deploy_test.go:116]: We still need a timeout for the test (should not let it run indefinitely). We could have a timed context with StateWaitMsg, but at that point we're achieving the same effect as here with more complicated code, so it's a matter of choice.

**Pull request merged**
